python/cudf/cudf/indexing.py

```diff
@@ -3,7 +3,43 @@
 from numba.cuda.cudadrv.devicearray import DeviceNDArray
 
 import cudf
-from cudf.utils.utils import is_single_value
+from cudf.utils.cudautils import arange
+from cudf.utils.utils import is_scalar
+
+
+def indices_from_labels(obj, labels):
+    from cudf.dataframe import columnops
+
+    labels = columnops.as_column(labels)
+
+    if pd.api.types.is_categorical_dtype(obj.index):
```

Comment:
There's a new faster `utils.dtypes.is_categorical_dtype` function we should use here instead of pandas'.

Reply:
Sweet!